static/app/components/searchSyntax/parser.tsx

```diff
@@ -699,13 +699,17 @@ export class TokenConverter {
   tokenValueText = (value: string, quoted: boolean) => {
     // We only want to consider a value to be `contains` if it is at least one character being wrapped in `*`
     const isContains = value.length > 2 && value.startsWith('*') && value.endsWith('*');
+    const isStartsWith = value.length > 2 && !isContains && value.endsWith('*');
+    const isEndsWith = value.length > 2 && !isContains && value.startsWith('*');
 
     return {
       ...this.defaultTokenFields,
       type: Token.VALUE_TEXT as const,
       value,
       quoted,
       contains: isContains,
+      startsWith: isStartsWith,
+      endsWith: isEndsWith,
```

Comment:
I think we need different names here, with no context these names mean nothing to me.

How about `hasLeadingWildcard` and `hasTrailingWildcard`. I also might suggest we rename `contains` to `isContains`

Reply:
Actually let's use an enum for this, kill contains, startsWith, and endsWith, since they're all mutually exlcusive we can have

```ts
wildcard: Wildcard.LEADING
wildcard: Wildcard.TRAILING
wildcard: Wildcard.SURRONDED
```

Reply:
Like this idea, went ahead and made the changes accordingly.